lib/box/enter.go

```diff
@@ -229,38 +207,107 @@ func Enter(dataDir string, cfg *ProcessConfig) error {
 		return trace.Wrap(err)
 	}
 
-	logrus.WithField("status", s).Info("Container process exited")
+	log.WithField("status", s).Info("Container process exited.")
 
 	if s != 0 {
 		return trace.Wrap(&ExitError{Code: s})
 	}
 	return nil
 }
 
-func setupIO(process *libcontainer.Process, rootuid, rootgid int, createtty bool) (*tty, error) {
-	if createtty {
-		t := &tty{}
+// EnterConfig specifies the configuration to execute a command inside the container
+type EnterConfig struct {
+	// Process specifies the process configuration to execute
+	Process ProcessConfig
+	// DataDir specifies the runc-specific data directory
+	DataDir string
+	// SELinux specifies whether SELinux support is on
+	SELinux bool
+}
+
+func (r *EnterConfig) checkAndSetDefaults() error {
+	if r.DataDir == "" {
+		r.DataDir = defaults.RuncDataDir
+	}
+	return nil
+}
+
+func (r *EnterConfig) seLinuxProcessLabel(rootfs string) (label string) {
+	if !selinux.GetEnabled() || !r.SELinux {
+		// Empty the label if SELinux has not been enabled
+		return ""
+	}
+	if r.Process.ProcessLabel == "" {
+		return getSELinuxProcLabel(rootfs, r.Process.Args[0])
+	}
+	// Use existing label if specified
+	return r.Process.ProcessLabel
+}
+
+func getContainer(dataDir string) (libcontainer.Container, error) {
+	factory, err := getLibContainerFactory(dataDir)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	absRoot, err := filepath.Abs(dataDir)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	files, err := ioutil.ReadDir(absRoot)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if len(files) == 0 {
+		return nil, trace.NotFound("planet container not found").AddField("data_dir", dataDir)
```

Comment:
Nit: I think we usually use dashes for multi-word fields, so I'd call this one "data-dir".